pkgs/by-name/be/beszel/package.nix

```diff
@@ -1,84 +1,89 @@
 {
-  buildGoModule,
+  stdenv,
   lib,
+  buildGoModule,
   fetchFromGitHub,
-  nix-update-script,
-  buildNpmPackage,
-  fetchpatch,
+  bun,
+  nodejs-slim,
 }:
-
 buildGoModule rec {
-  pname = "beszel";
-  version = "0.6.2";
 
-  src = fetchFromGitHub {
+  pname = "beszel-hub";
+  version = "0.7.3";
+
+  repo = fetchFromGitHub {
     owner = "henrygd";
     repo = "beszel";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-x9HU+sDjxRthC4ROJaKbuKHPHgxFSpyn/dywyGWE/v8=";
+    rev = "v${version}";
+    hash = "sha256-BGwCfWDgLbcyF9yHlTlvbGKQ11HfO+VKcPt83htUDxA=";
   };
 
-  webui = buildNpmPackage {
-    inherit
-      pname
-      version
-      src
-      meta
-      ;
-
-    npmFlags = [ "--legacy-peer-deps" ];
-
-    patches = [
-      # add missing @esbuild for multi platform
-      # https://github.com/henrygd/beszel/pull/235
-      # add missing @esbuild for multi platform
-      # https://github.com/henrygd/beszel/pull/235
-      ./0001-fix-build.patch
+  webui = stdenv.mkDerivation {
+    pname = "beszel-hub-webui";
+    inherit version;
+    src = "${repo}/beszel/site";
+    depsBuildBuild = [
+      bun
+      nodejs-slim
     ];
 
-    buildPhase = ''
-      runHook preBuild
+    configurePhase = ''
```

Comment:
I did consider using `buildNpmPackage`, but upstream builds it via shell commands in their [Makefile](https://github.com/henrygd/beszel/blob/v0.7.4/beszel/Makefile)... isn't it better™ if we build like they do?

That said, it's like I'm opposed to switch to `buildNpmPackage` if that provides any benefits (and if you can vouch it's the same as the commands in the Makefile - I myself don't know much about Javascript so I wouldn't know).

Can you elaborate?

Reply:
I am not opposed in principle :)

Could you explain what's the benefit and provide assurance that `buildNpmPackage` is equivalent to the Makefile upstream is using?

Reply:
Could you explain why you changed their old code, but doesn't bring any substantial improvement?

Adding makefile, just, build.sh or something else upstream is just for their convenience. Of course we should also think in a way that is convenient for us, as long as we don't break installation.

In addition, for other people's code, unless they no longer maintain it, or to improve correctness, do not make useless changes to it. This is not a respectful way to work together.

Reply:
Sorry @Aleksanaa I've only seen your message now, after closing this PR to send one for beszel 0.8.0. I'll report your comment and reply there.

Reply:
@Aleksanaa on second thought, I won't post a new PR.

If you read the whole discussion from the top, you'll see that I did explain the reason for everything I did, and that the same can't be said for the rework I've been asked to do.

I must say that I am really surprised that anyone who had read the discussion in this PR (or even just this thread that starts with someone comment someone else's work with "this is stupid") might think that *I* may be the one not being respectful.

Having said that for the record, I am not going to make a big deal out of this and, in all honesty, this discussion has already been much more of a hassle for me than I care about contributing a package to nixos.

I am sure everyone involved in this discussion did their best and I truly believe the discussion ended up being so unproductive because of accidental misunderstandings.

I am also sure the three of us will be able to collaborate with no hitches the next time the opportunity arises.